multiversx/mx-chain-go
# Fix /transaction/pool with nonce-gaps

## Reasoning behind the pull request
- when calling /transaction/pool with nonce-gaps, it does not check sender's account nonce, thus leading to missing the first gap that led to accumulating txs in pool
  
## Proposed changes
- check sender's account nonce as well
- apply ToLower on fields

## Testing procedure
- system test
- scenario:
   - generate some txs with different nonces from a specific sender, but make sure to break the nonces so they go to pool
   - call `/address/erd1.../nonce`, where erd1... is the sender address used in the above step
   - **on a node that is in the same shard with the sender**(or on a proxy, but make sure the proxy is started with `AllowEntireTxPoolFetch = true` in config.toml) call `/transaction/pool?by-sender=erd1...&nonce-gaps=true` to get the nonce gaps for this account
   - make sure that the gap starting from the sender's account nonce to the first nonce in pool is also returned by the transactions pool api
   - make sure following calls work also:
      - `/transaction/pool?by-sender=erd1...&fields=gasprice`
      -  `/transaction/pool?by-sender=erd1...&fields=GASPRICE`
      -  `/transaction/pool?by-sender=erd1...&fields=gasPrice`

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
